tests/smolyay_test.py

```diff
@@ -0,0 +1,159 @@
+import pytest
+import numpy
+from smolyay.smolyak import IndexGrid,generate_compositions
+
+@pytest.fixture
+def expected_points():
+    """grid_point_indexes for dimension=2,exactness=2,index_per_level=[1,2,2]"""
+    return numpy.array([[0,0],[0,1],[0,2],[0,3],[0,4],[1,0],
+        [1,1],[1,2],[2,0],[2,1],[2,2],[3,0],[4,0]], dtype=numpy.int32)
+
+@pytest.fixture
+def expected_indexes():
+    """level_indexes for dimension=2,exactness=2,index_per_level=[1,2,2]"""
+    return [[0],[1,2],[3,4]]
+
+def test_initial_dimension():
+    """test initialized class returns correct dimension"""
+    test_class = IndexGrid(1,2,[1,2,3])
+    assert test_class.dimension == 1
+
+def test_initial_exactness():
+    """test initialized class returns correct exactness"""
+    test_class = IndexGrid(1,2,[1,2,3])
+    assert test_class.exactness == 2
+
+def test_initial_index_by_level():
+    """test initialized class returns correct index by level"""
+    test_class = IndexGrid(1,2,[1,2,3])
+    assert numpy.array_equiv(test_class.index_per_level,[1,2,3])
```

Comment:
I think that `array_equiv` is a weaker test so this should be `array_equal`. Was it failing otherwise?
```suggestion
    assert numpy.array_equal(test_class.index_per_level,[1,2,3])
```
In general, these tests make clear that there is a mixture of lists and NumPy arrays being used in the class. Do we want to perhaps use only lists? That would allow these tests to simplify to using `==` operators, and I could imagine that being helpful elsewhere. It should be as easy as converting the NumPy array `tolist` before storing it in the class.